### PR TITLE
Use more smart ptrs and u8strings

### PR DESCRIPTION
**Pull request by rzhao271**
Ref #11159

### CODE REVIEW

File: src/openrct2/actions/ChangeMapSizeAction.cpp
Comment (duncanspumpkin) [inline src/openrct2/actions/ChangeMapSizeAction.cpp:71]: Keep the *.

**Review by duncanspumpkin — APPROVED** [review]

File: src/openrct2/ui/DummyUiContext.cpp
Comment (Broxzier) [inline src/openrct2/ui/DummyUiContext.cpp:191]: `std::unique_ptr::get` should never be exposed outside functions. To prevent the pointer from being stored and used elsewhere, return a reference to the `std::unique_ptr` instead. You'd need to change this in `UiContext.h` too.

```suggestion
        std::unique_ptr<IWindowManager>& GetWindowManager() override
```

File: src/openrct2-ui/UiContext.cpp
Comment (Broxzier) [inline src/openrct2-ui/UiContext.cpp:678]: Same as previous note.

**Review by Broxzier — DISMISSED** [review]

File: src/openrct2/ui/DummyUiContext.cpp
Comment (duncanspumpkin) [reply src/openrct2/ui/DummyUiContext.cpp:191]: It is pretty standard practice to return the pointer and not a unique ptr ref and we do do this in other bits of the code.

Comment (rzhao271) [reply src/openrct2/ui/DummyUiContext.cpp:191]: Yeah I don't think I've seen unique pointer refs before for accessors. Chromium also uses raw pointers in that case https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/render_frame_host_impl.cc;l\=2142?q=render_frame_host&ss=chromium.
The assumption is that when a raw pointer is returned, that already indicates lack of ownership. Otherwise the pointer should've been moved or released.

Comment (Broxzier) [reply src/openrct2/ui/DummyUiContext.cpp:191]: I don't see an advantage of returning raw pointers over `std::unique_ptr` references (or even better `T&` if you know it's not null and not going to change). With raw pointers you don't know the lifetime and can accidentally move  or release them, both of which will not be easily done by accident if you return `std::unique_ptr<T>&`. There is also no performance overhead whatsoever (unlike with shared_ptr).

Comment (duncanspumpkin) [reply src/openrct2/ui/DummyUiContext.cpp:191]: Well the issue is you could call release on a unique ptr ref. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-ptr core guidelines also agrees that the standard is a raw pointer is not owning.

Comment (rzhao271) [reply src/openrct2/ui/DummyUiContext.cpp:191]: `T&` does look good to me, though. Will have to confirm later whether I can use that instead.

Comment (Broxzier) [reply src/openrct2/ui/DummyUiContext.cpp:191]: Since this PR moves more towards these guidelines, I guess it's fine. I won't block this PR just for this, since the rest looks fine to me. I'll leave it to you to decide if you want to change this.

Comment (rzhao271) [reply src/openrct2/ui/DummyUiContext.cpp:191]: Hm, maybe not in this PR. It looks like it'll cause a ripple effect throughout the codebase because `->` would have to be changed to `.` as well:
Screenshot showing all the places that use the GetWindowManager accessor

**Pull request merged**
